realm/realm-library/src/androidTestObjectServer/kotlin/io/realm/SessionTests.kt

```diff
@@ -0,0 +1,562 @@
+import kotlin.test.assertFalse
+import kotlin.test.assertTrue
+import kotlin.test.fail
+
+@RunWith(AndroidJUnit4::class)
+class SessionTests {
+    private lateinit var configuration: SyncConfiguration
+    private lateinit var app: TestRealmApp
+    private lateinit var user: RealmUser
+
+    @get:Rule
+    val configFactory = TestSyncConfigurationFactory()
+
+    private val looperThread = BlockingLooperThread()
+
+    @Before
+    fun setUp() {
+        Realm.init(InstrumentationRegistry.getInstrumentation().targetContext)
+        app = TestRealmApp()
+        // TODO We could potentially work without a fully functioning user to speed up tests, but
+        //  seems like the old  way of "faking" it, does now work for now, so using a real user.
+        // user = SyncTestUtils.createTestUser(app)
+        user = createNewUser(app)
+        configuration = SyncConfiguration.defaultConfig(user, "default")
+    }
+
+    @After
+    fun tearDown() {
+        if (this::app.isInitialized) {
+            app.close()
+        }
+    }
+
+    @Test
+    fun get_syncValues() {
+        val session = SyncSession(configuration)
+        Assert.assertEquals("ws://127.0.0.1:9090/", session.serverUrl.toString())
+        Assert.assertEquals(user, session.user)
+        Assert.assertEquals(configuration, session.configuration)
+    }
+
+    @Test
+    fun addDownloadProgressListener_nullThrows() {
+        val session = app.sync.getOrCreateSession(configuration)
+        try {
+            session.addDownloadProgressListener(ProgressMode.CURRENT_CHANGES, TestHelper.getNull())
+            Assert.fail()
+        } catch (ignored: IllegalArgumentException) {
+        }
+    }
+
+    @Test
+    fun addUploadProgressListener_nullThrows() {
+        val session = app.sync.getOrCreateSession(configuration)
+        try {
+            session.addUploadProgressListener(ProgressMode.CURRENT_CHANGES, TestHelper.getNull())
+            Assert.fail()
+        } catch (ignored: IllegalArgumentException) {
+        }
+    }
+
+    @Test
+    fun removeProgressListener() {
+        val realm = Realm.getInstance(configuration)
+        val session = app.sync.getOrCreateSession(configuration)
+        val listeners = arrayOf(
+                null,
+                ProgressListener { progress: Progress? -> },
+                ProgressListener { progress: Progress? -> }
+        )
+        session.addDownloadProgressListener(ProgressMode.CURRENT_CHANGES, TestHelper.allowNull(listeners[2]))
+        session.addDownloadProgressListener(ProgressMode.CURRENT_CHANGES, TestHelper.allowNull(listeners[2]))
+
+        // Check that remove works unconditionally for all input
+        for (listener in listeners) {
+            session.removeProgressListener(TestHelper.allowNull(listener))
+        }
+        realm.close()
+    }
+
+    // Check that a Client Reset is correctly reported.
+    @Test
+    fun errorHandler_clientResetReported() = looperThread.runBlocking {
```

Comment:
I suspect all of these ClientReset tests needs to be ignored for now, but a pleasant surprise if not.

Reply:
Seems to work.